bolt-sidecar/src/telemetry/metrics.rs

```diff
@@ -0,0 +1,153 @@
+use std::time::Duration;
+
+use metrics::{counter, describe_counter, describe_gauge, describe_histogram, gauge, histogram};
+use reth_primitives::TxType;
+
+use crate::primitives::tx_type_str;
+
+/// Metrics for the commitments API.
+#[derive(Debug, Clone, Copy)]
+pub struct ApiMetrics;
+
+#[allow(missing_docs)]
+impl ApiMetrics {
+    pub fn start() {
+        ApiMetricType::describe_all();
+    }
+
+    /// Counters ----------------------------------------------------------------
+
+    pub fn increment_total_http_requests(method: String, path: String, status: String) {
```

Comment:
quick question: does this work with &str or only String as arguments? I think it would be more ergonomic to use with slices if that's doable. If not, that's ok

Reply:
No in general labels must be owned values. However you're right that a &str signature is more idiomatic. Changing it rn

Reply:
yup that's perfectly fine, thanks